src/instructlab/rag/document_store_factory.py

```diff
@@ -0,0 +1,65 @@
+# Standard
+from typing import cast
+import importlib
+import logging
+
+# First Party
+from instructlab.rag.document_store import DocumentStoreIngestor, DocumentStoreRetriever
+from instructlab.rag.rag_configuration import (
+    DocumentStoreConfig,
+    DocumentStoreType,
+    EmbeddingModelConfig,
+    RetrieverConfig,
+)
+
+logger = logging.getLogger(__name__)
+
+
+def create_document_retriever(
+    type: DocumentStoreType,
+    document_store_config: DocumentStoreConfig,
+    retriever_config: RetrieverConfig,
+) -> DocumentStoreRetriever:
+    try:
+        module_name = "instructlab.rag.haystack.document_store_factory"
+        module = importlib.import_module(module_name)
+
+        function_name = f"create_{type.value}_document_retriever"
+        function = getattr(module, function_name)
```

Comment:
I would still argue that, as a general policy, levels of abstraction such as this one are best not implemented until actually needed. Maybe the next document store required will not be supported in Haystack, for example. 

Code is a liability; less is better. I can see the argument that we expect to support multiple document stores, but right now that is simply not required, and the focus is on getting the simplest possible vertical slice working.

This is especially the case when getting into manually importing modules using string identifiers and `getattr` loading functions - these things can be very fragile.

Reply:
👍 I will remove both the function lookup and the store type enum, providing an hardcoded, non-configurable implementation based on milvus.